### PR TITLE
Bjorncs/scale threadpools locally

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @bratseth

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/vespa/model/container/xml/SearchHandler.java [hunk 1]
```diff
@@ -48,22 +48,10 @@ public void getConfig(ContainerThreadpoolConfig.Builder builder) {
...
-                // Controls max number of concurrent requests per container
-                int workerThreads = Math.max(8, (int)Math.ceil(vcpu * 2.0));
-                builder.maxThreads(workerThreads);
-                builder.minThreads(workerThreads);
-
-                // This controls your burst handling capability.
-                // 0 => No extra burst handling beyond you max concurrent requests (maxthreads).
-                // N => N times max concurrent requests as a buffer for handling bursts
-                builder.queueSize((int)(workerThreads * 40.0));
+                builder.maxThreads(500).minThreads(500).queueSize(0);
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/vespa/model/container/xml/SearchHandler.java:54]: Use -4 for non hosted, but same approach.

File: container-core/src/main/java/com/yahoo/container/handler/threadpool/DefaultContainerThreadpool.java [hunk 2]
```diff
@@ -74,24 +86,26 @@ private void closeInternal() {
...
+        else threads = Math.abs(config.minThreads()) * cpus;
+        return Math.min(threads, max);
     }
 
-    private static int computeCoreThreadPoolSize(int corePoolSize, int maxNumThreads) {
-        return Math.min(
-                corePoolSize <= 0 ? Runtime.getRuntime().availableProcessors() * 2 : corePoolSize,
-                maxNumThreads);
+    private int queueSize(ContainerThreadpoolConfig config, int maxThreads) {
+        return config.queueSize() >= 0 ? config.queueSize() : Math.max(650, Math.abs(config.queueSize()) * maxThreads);
```
Comment (baldersheim) [inline container-core/src/main/java/com/yahoo/container/handler/threadpool/DefaultContainerThreadpool.java:108]: Use a named constant that gives a hint to purpose.

**Review by baldersheim — CHANGES_REQUESTED** [review]
Fix the comments and you should be good.

**Comment by bjorncs** [comment]
@baldersheim PTAL

**Review by baldersheim — APPROVED** [review]

**Pull request merged**